docs/source/nlp/api.rst

```diff
@@ -1,99 +1,141 @@
-NeMo NLP collection API
+NeMo Megatron API
 =======================
 
-Model Classes
--------------
+Pretraining Model Classes
+-------------------------
+
+.. autoclass:: nemo.collections.nlp.models.language_modeling.megatron_base_model.MegatronBaseModel
+    :show-inheritance: 
+    :no-members:
+    :members: __init__, configure_optimizers
+
+.. autoclass:: nemo.collections.nlp.models.language_modeling.megatron_gpt_model.MegatronGPTModel
+    :show-inheritance:
+    :no-members:
+    :members: generate, training_step, validation_step, build_train_valid_test_datasets, setup, on_save_checkpoint, on_load_checkpoint
+
+.. autoclass:: nemo.collections.nlp.models.language_modeling.megatron_bert_model.MegatronBertModel
+    :show-inheritance: 
+    :no-members:
+    :members: training_step, validation_step, build_train_valid_test_datasets, build_LDDL_data, setup, on_save_checkpoint, on_load_checkpoint
+
+.. autoclass:: nemo.collections.nlp.models.language_modeling.megatron_bart_model.MegatronBARTModel
+    :show-inheritance: 
+    :no-members:
+    :members: training_step, validation_step, build_train_valid_test_datasets, build_LDDL_data, setup, on_save_checkpoint, on_load_checkpoint
+
+.. autoclass:: nemo.collections.nlp.models.language_modeling.megatron_retrieval_model.MegatronRetrievalModel
+    :show-inheritance: 
+    :no-members:
+    :members: generate, training_step, validation_step, build_train_valid_test_datasets, setup
+
+.. autoclass:: nemo.collections.nlp.models.language_modeling.megatron_t5_model.MegatronT5Model
+    :show-inheritance: 
+    :no-members:
+    :members: complete, encode, decode, add_special_tokens_to_tokenizer, training_step, validation_step, build_train_valid_test_datasets, setup
+
+Customization Model Classes
+---------------------------
+
+.. autoclass:: nemo.collections.nlp.models.language_modeling.megatron_gpt_sft_model.MegatronGPTSFTModel
+    :show-inheritance: 
+    :no-members:
+    :members: generate, training_step, validation_step, build_train_valid_test_datasets, setup
```

Comment:
Should just include methods from parent classes? If not, training_step comes from the parent GPTModel class.